plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -654,30 +668,57 @@ public List<Setting<?>> getSettings() {
                 MLCommonsSettings.ML_COMMONS_MODEL_ACCESS_CONTROL_ENABLED,
                 MLCommonsSettings.ML_COMMONS_CONNECTOR_ACCESS_CONTROL_ENABLED,
                 MLCommonsSettings.ML_COMMONS_TRUSTED_CONNECTOR_ENDPOINTS_REGEX,
-                MLCommonsSettings.ML_COMMONS_MEMORY_FEATURE_ENABLED
+                MLCommonsSettings.ML_COMMONS_MEMORY_FEATURE_ENABLED,
+                MLCommonsSettings.ML_COMMONS_RAG_PIPELINE_FEATURE_ENABLED
             );
         return settings;
     }
 
+    /**
+     *
+     * Search processors for Retrieval Augmented Generation
+     *
+     */
+
     @Override
     public List<SearchPlugin.SearchExtSpec<?>> getSearchExts() {
-        return List
-            .of(
+        List<SearchPlugin.SearchExtSpec<?>> searchExts = new ArrayList<>();
+
+        searchExts
+            .add(
                 new SearchPlugin.SearchExtSpec<>(
                     GenerativeQAParamExtBuilder.PARAMETER_NAME,
                     input -> new GenerativeQAParamExtBuilder(input),
                     parser -> GenerativeQAParamExtBuilder.parse(parser)
                 )
             );
+
+        return searchExts;
     }
 
     @Override
     public Map<String, Processor.Factory<SearchRequestProcessor>> getRequestProcessors(Parameters parameters) {
-        return Map.of(GenerativeQAProcessorConstants.REQUEST_PROCESSOR_TYPE, new GenerativeQARequestProcessor.Factory());
+        Map<String, Processor.Factory<SearchRequestProcessor>> requestProcessors = new HashMap<>();
+
+        requestProcessors
+            .put(
+                GenerativeQAProcessorConstants.REQUEST_PROCESSOR_TYPE,
+                new GenerativeQARequestProcessor.Factory(() -> this.ragSearchPipelineEnabled)
```

Comment:
The problem with sending this boolean is when the change in the setting will happen it will not be propagated to the Processor because factory is created once.

Reply:
I am passing a Supplier through the Factory constructor so it can be accessed in the Factory.create() method.

Reply:
I think you can just abstract this flag in this class: https://github.com/opensearch-project/ml-commons/blob/f88e18c5873c9d7e198f1137b7269364bcd5c1bd/plugin/src/main/java/org/opensearch/ml/settings/MLFeatureEnabledSetting.java#L15-L34

and then in the processor directly call the functions of the MLFeatureEnabledSetting class.

Reply:
and this setting class is already getting init in the createComponents method here: https://github.com/opensearch-project/ml-commons/blob/86eb953fe6b49aac0646307d10ce9f9b36af01f5/plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java#L337

so rather than passing this consumer you can pass the instance of MLFeatureEnabledSetting class to the processor factory. Also this instance is a singleton instance.

Reply:
Yeah, @ylwu-amzn @jngz-es @dhrubo-os know about this.  You were not part of the earlier discussion, but this new class is in 2.x, but not in the feature branch we're currently working off of.

Reply:
You can see in my comments that we plan to use this once we are merged into 2.x.

Reply:
got it. If that is the plan.